File: shade-backend/.claude/worktrees/brave-heyrovsky-ffb76d/internal/dto/auth_dto.go
package dto

type RegisterRequest struct {
	IdentityPublicKey             string `json:"identity_public_key"`
	EncryptedIdentityPrivateKey   string `json:"encrypted_identity_private_key"`
	EncryptionPublicKey           string `json:"encryption_public_key"`
	EncryptedEncryptionPrivateKey string `json:"encrypted_encryption_private_key"`
	Salt                          string `json:"salt"`
	DeviceModel                   string `json:"device_model"`
	FCMToken                      string `json:"fcm_token"`
}

type RegisterResponse struct {
	CoreGuardID string `json:"core_guard_id"`
	UserID      string `json:"user_id"`
	Message     string `json:"message"`
}

type LoginInitRequest struct {
	CoreGuardID string `json:"core_guard_id"`
}

type LoginInitResponse struct {
	EncryptedIdentityPrivateKey string `json:"encrypted_identity_private_key"`
	EncryptedEncryptionPrivateKey string `json:"encrypted_encryption_private_key"`
	Salt                        string `json:"salt"`
	Challenge                   string `json:"challenge"`
}

type LoginVerifyRequest struct {
	CoreGuardID string `json:"core_guard_id"`
	Challenge   string `json:"challenge"`
	Signature   string `json:"signature"`
	DeviceModel string `json:"device_model"`
	FCMToken    string `json:"fcm_token"`
}

type LoginVerifyResponse struct {
	CoreGuardID string `json:"core_guard_id"`
	UserID      string `json:"user_id"`
	AccessToken string `json:"access_token"`
	Message     string `json:"message"`
}

File: shade-android-main/app/src/main/java/com/shade/app/data/remote/dto/AuthDto.kt
package com.shade.app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class RegisterRequest(
    @SerializedName("identity_public_key") val identityPublicKey: String,
    @SerializedName("encrypted_identity_private_key") val encryptedIdentityPrivateKey: String,
    @SerializedName("encryption_public_key") val encryptionPublicKey: String,
    @SerializedName("encrypted_encryption_private_key") val encryptedEncryptionPrivateKey: String,
    @SerializedName("salt") val salt: String,
    @SerializedName("device_model") val deviceModel: String,
    @SerializedName("fcm_token") val fcmToken: String
)

data class RegisterResponse(
    @SerializedName("core_guard_id") val shadeId: String,
    @SerializedName("user_id") val userId: String?,
    @SerializedName("device_id") val deviceID: String,
    @SerializedName("message") val message: String
)

data class LoginInitRequest(
    @SerializedName("core_guard_id") val shadeId: String,
)

data class LoginInitResponse(
    @SerializedName("encrypted_identity_private_key") val encryptedIdentityPrivateKey: String,
    @SerializedName("encrypted_encryption_private_key") val encryptedEncryptionPrivateKey: String,
    @SerializedName("salt") val salt: String,
    @SerializedName("challenge") val challenge: String
)

data class LoginVerifyRequest(
    @SerializedName("core_guard_id") val shadeId: String,
    @SerializedName("challenge") val challenge: String,
    @SerializedName("signature") val signature: String,
    @SerializedName("device_model") val deviceModel: String,
    @SerializedName("fcm_token") val fcmToken: String,
    @SerializedName("device_id") val deviceId: String? = null
)

data class LoginVerifyResponse(
    @SerializedName("core_guard_id") val shadeId: String,
    @SerializedName("user_id") val userId: String,
    @SerializedName("access_token") val accessToken: String,
    @SerializedName("refresh_token") val refreshToken: String,
    @SerializedName("device_id") val deviceId: String,
    @SerializedName("message") val message: String
)

data class RefreshRequest(
    @SerializedName("refresh_token") val refreshToken: String,
)

data class RefreshResponse(
    @SerializedName("access_token") val accessToken: String,
    @SerializedName("refresh_token") val refreshToken: String,
)
